fedimint-rocksdb/src/lib.rs

```diff
@@ -51,6 +51,27 @@ impl From<RocksDb> for rocksdb::OptimisticTransactionDB {
     }
 }
 
+// When finding by prefix iterating in Reverse order, we need to start from
+// "prefix+1" instead of "prefix", using lexicographic ordering. See the tests
+// below
+fn next_prefix(prefix: &[u8]) -> Result<Vec<u8>> {
+    let mut next_prefix = prefix.to_vec();
```

Comment:
Thought - this fn could be performance sensitive ... maybe? ... slightly?

I wonder if passing `prefix: &mut [u8]` to avoid this clone wouldn't make sense?

Reply:
This is called just once per find by prefix and has only one or two bytes.

It won't really avoid cloning because we need the original prefix too.

Reply:
> It won't really avoid cloning because we need the original prefix too.

Caller can do it if they need to, and it makes the code here easier, IMO. But not a blocker.

Reply:
Code gets ugly if you change that